builders/interact/ethereum-api/precompiles/call-permit.md

```diff
@@ -0,0 +1,275 @@
+3. The domain separator will be created using `"Call Permit Precompile"` exactly for the name, the version of your DApp or platform, the chain ID of the network the signature is to be used on, and the address of the contract that will verify the signature. Note that you'll need to specify the chain ID of your ContainerChain in the script to generate the correct signature
+4. All of the assembled data, the `types`, `domain`, `primaryType` and `message`, will be signed using MetaMask (either in the browser or through the MetaMask's JavaScript signing library)
+5. The signature will be returned and you can use [Ethers.js](https://docs.ethers.org/){target=\_blank} [`Signature.from` method](https://docs.ethers.org/v6/api/crypto/#Signature_from){target=\_blank} to return the `v`, `r`, and `s` values of the signature
+
+### The Call Permit Arguments {: #call-permit-arguments }
+
+As seen in the [Call Permit Interface](#the-call-permit-interface) section, the `dispatch` function takes the following parameters: `from`, `to`, `value`, `data`, `gasLimit`, `deadline`, `v`, `r`, and `s`.
+
+In order to get the signature arguments (`v`, `r`, and `s`), you'll need to sign a message containing the arguments for the remainder of the aforementioned parameters, plus the nonce of the signer.
+
+- `from` - the address of the account you want to sign the call permit with
+- `to` - the contract address for the `SetMessage.sol` contract
+- `value` - can be `0` for this example as you'll just be setting a message instead of transferring any funds
+- `data` - you can send any message you would like. You'll just need the hex representation of the message you want to set using the `SetMessage.sol` contract. This will contain the function selector of the `set` function and the string of the message. For this example, you can send `hello world`. To do so, you can use this hex representation:
+     ```text
+     0x4ed3885e0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000b68656c6c6f20776f726c64000000000000000000000000000000000000000000
+     ```
+- `gasLimit` - `100000` will be enough to send the dispatched call
+- `deadline` - you can get the current time in UNIX seconds by running `console.log(Date.now())` in a JavaScript script or a browser console. Once you have the current time, you should generously add additional seconds to represent when the call permit will expire
+
+The nonce of the signer will also be needed. If this is your first time signing a call permit the nonce will be `0`. You can also check the nonce in Remix:
+
+1. Expand the call permit contract
+2. Next to the **nonces** function, enter the address of the signer and click on **nonces**
+3. The result will be returned directly under the function
+
+![Get the nonce](/images/builders/interact/ethereum-api/precompiles/call-permit/call-6.webp)
+
+### Use MetaMask's JS Signing Library {: #use-metamasks-signing-library }
```

Comment:
I would actually love to get rid of even using MetaMask, and simplify it with just using Ethers. There is an example in this tutorial: https://docs.moonbeam.network/tutorials/eth-api/call-permit-gasless-txs/#build-an-eip-712-signed-message

Reply:
Done